### ISSUE TITLE
Docker for Mac 3.5.2 starting IntelliJ IDEA on Dashboard and About

### ISSUE THREAD

**Issue by temp** [docker/for-mac#5840]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: E815ABA4-EB44-4536-9A0A-E3069FFCF0BC/20210713201258

### Expected behavior

Docker for Mac simply showing the Dashboard or the About Dialog.

### Actual behavior

Docker for Mac shows the Dashboard or the About Dialog, but also starts IntelliJ IDEA. Every time.
IDEA then shows this error:
```Can not open file /Users/<me>/Library/Containers/com.docker.docker/Data/--list-extensions```

### Information
It is reproducable, it happens every time I click on Tray Icon / Dashboard or Tray Icon / About Docker Desktop
My Macbook is a Intel MacBook Pro

### Steps to reproduce the behavior

Click on Tray Icon / Dashboard or Tray Icon / About Docker Desktop

![Bildschirmfoto 2021-07-13 um 22 11 07](https://user-images.githubusercontent.com/216128/125518987-6d906c01-9a03-4b91-94c7-69566324276a.png)
![Bildschirmfoto 2021-07-13 um 22 11 38](https://user-images.githubusercontent.com/216128/125519000-8476f04d-b4d0-440c-8822-9b6ba8a41bd5.png)

**Comment by temp** [comment]
Addendum: This did *not* happen before 3.5.0, maybe 3.5.1, and I didn't use IntelliJ IDEA for half a year, so I'd say it didn't happen while using IDEA.

**Comment by rumpl** [comment]
@temp hi, thank you for opening this issue

Do you happen to have an alias called `code` in your `$PATH` that points to Intellij?

It would also be helpful for us if you gave us the output of this command: 

```
mdfind kMDItemCFBundleIdentifier=com.microsoft.VSCode
```

Thanks

**Comment by temp** [comment]
Alias: doesn't seem so... And mdfind returns nothing:

```
$ code
bash: code: command not found
```
```
$ alias | grep code
$
```
```
$ mdfind kMDItemCFBundleIdentifier=com.microsoft.VSCode
$
```

**Comment by rumpl** [comment]
@temp Would you mind having a quick zoom call with me? You can contact me at djordje.lukic@docker.com

**Comment by temp** [comment]
@rumpl done, check your mail!

**Comment by temp** [comment]
@rumpl Here is a screenshot from the activity monitor info of the intellij idea process:

![Bildschirmfoto 2021-07-15 um 16 05 57](https://user-images.githubusercontent.com/216128/125801701-278b25c0-7218-4207-b8e5-e968d5d97c9e.png)

The parent process is `com.docker.dev-envs`

**Comment by troy-folger-sp** [comment]
Same thing happening to me. Also exact same results as above for code, alias, mdfind.

**Comment by temp** [comment]
At least I'm not alone :-D

**Comment by vorsprung** [comment]
Me also, Docker Desktop 3.5.2.18 on MacOS Catalina 10.15.7
Also an error with --list-extensions

**Comment by vorsprung** [comment]
removing the docker plugin thing from intellij did not help

**Comment by boredelf** [comment]
Hi, guys!

Same issue here. I tried renaming my `/usr/local/bin/idea`, causing Docker Desktop to complain and ask to reset to factory defaults, which I did, expecting the problem to go away, but it didn't. I also tried uninstalling both of them and cleaning their related directories (in `Library/Application Support`), to no avail.

Same MacOS and Docker Desktop versions as @vorsprung.

Any help would be greatly appreciated. Cheers! :)

**Comment by wayne311** [comment]
I am also having the exact same issues, Same OS and same errors.

**Comment by theartofdevel** [comment]
same problem. 
i can zoom call any time to help fix this problem

Telegram: @dikkini

**Comment by rumpl** [comment]
@theartofdevel thanks! Contact me by email

**Comment by shikimoon** [comment]
the same problem，Docker Desktop 3.5.2 on MacOS Catalina 10.15.7

**Comment by BrucessKING** [comment]
After I upgraded IntelliJ idea to 2021.2, the error message disappeared. IntelliJ idea can open --list-extensions normally, but I still don't understand why this file is opened

**Comment by boredelf** [comment]
> After I upgraded IntelliJ idea to 2021.2, the error message disappeared. IntelliJ idea can open --list-extensions normally, but I still don't understand why this file is opened

Same here. :(

**Comment by andreas-trvlk** [comment]
I can confirm that the issue starts happening on Desktop version 3.5.2. Downgrading to version 3.5.1 resolves the issue. I thought it was Intellij issue, but I am sure it is docker desktop issue

**Comment by theartofdevel** [comment]
after a few tests, me and developer confirmed, that issue in docker desktop and it connects with docker environment and VS Code. If you install VS Code than issue will be fixed.

**Comment by jtkendall** [comment]
Installing VS Code does in fact stop the issue, however I shouldn't have to install an app I don't use to keep Docker for Mac from opening IntelliJ IDEA.

**Comment by BrucessKING** [comment]
I just want to install docker. Why do you want me to install vs code, a software I don't need? This is not the right way to solve the issue.

**Comment by vorsprung** [comment]
> after a few tests, me and developer confirmed, that issue in docker desktop and it connects with docker environment and VS Code. If you install VS Code than issue will be fixed.

Not in my case, I have a fully up to date VS Code installed and docker desktop still starts intellij every time it starts up

**Comment by theartofdevel** [comment]
also, you have to install plugin, but i forget which one :(
anyway, developer promise to fix this issue in next one release

**Comment by nmfzone** [comment]
Just want to notify if it's also happened on me..

Docker Desktop version 3.5.2.18 on MacOS Catalina.

It's little annoying since the previous version is fine..

**Comment by andreas-trvlk** [comment]
Is this issue fixed in 3.6.0?

**Comment by YGolikov** [comment]
> Is this issue fixed in 3.6.0?

yes

**Comment by temp** [comment]
No, it's not. 3.6.0, and IntelliJ Idea is still starting on Dashboard. It didn't start it on About Us.
And it seems like it does this only once now.

**Comment by GKTheOne** [comment]
Still with 4.1.1 ?

On the Docker for Mac dashboard, every time I move the view to `Dev Environments`, IDEA will open `--list-extensions` in LightEdit mode.

**Comment by eeckhardt-chwy** [comment]
Same thing happens to me. Every time Docker Desktop comes up asking to update and I let it do so, it loads IntelliJ in LightEdit mode with blank file `--list-extensions` open (which seems like a switch gone wrong rather than a real file expected to be on the filesystem).

It honestly seems like the developers could search for the string `--list-extensions` in their codebase and figure out why that is coming in as a filename instead of as a switch. A shell command gone wrong? A missing space character somewhere? An accidental miss on putting double quotes around an environment variable in a shell script?

I grepped my entire hard disk for `--list-extensions` and found an IntelliJ log indicating this has happened to me 21 times, `./Library/Logs/JetBrains/IntelliJIdea2021.2/idea.log:--list-extensions`.

Due diligence:
```
/ % code
zsh: command not found: code
/ % which code
code not found
/ % alias | grep code
/ % mdfind kMDItemCFBundleIdentifier=com.microsoft.VSCode

/ %
```
macOs: Big Sur Version 11.6 MacBook Pro (16-inch, 2019) Processor Intel Core i7
Docker Desktop for Mac: Version 3.5.2.18

**Comment by glours** [comment]
Hello everyone,

Last week we released the version `4.2.0` of Docker Desktop, in this one we fixed a bug in Docker Dev Environment which should be the source of this issue. 

Can you try it and let me know if you still have this weird behaviour?

**Comment by GKTheOne** [comment]
The update fixed it for me 👍
